logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/user/IsSelfATeamMemberUseCase.kt

```diff
@@ -1,33 +1,19 @@
-/*
- * Wire
- * Copyright (C) 2023 Wire Swiss GmbH
- *
- * This program is free software: you can redistribute it and/or modify
- * it under the terms of the GNU General Public License as published by
- * the Free Software Foundation, either version 3 of the License, or
- * (at your option) any later version.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with this program. If not, see http://www.gnu.org/licenses/.
- */
-
 package com.wire.kalium.logic.feature.user
 
 import com.wire.kalium.logic.feature.SelfTeamIdProvider
 import com.wire.kalium.logic.functional.fold
 
+fun interface IsSelfATeamMemberUseCase {
+    suspend operator fun invoke(): Boolean
+}
+
 /**
- * Return if self user is part of a team or noz
+ * Return if self user is part of a team or not
  */
-class IsSelfATeamMemberUseCase internal constructor(
+internal class IsSelfATeamMemberUseCaseImpl internal constructor(
     private val selfTeamIdProvider: SelfTeamIdProvider
```

Comment:
Same here, we can inject the dispatchers and gain some standardization on our Use Cases